libs/components/forms/src/lib/modules/file-attachment/file-attachment/file-attachment.component.scss

```diff
@@ -1,120 +1,223 @@
+  --sky-override-file-attachment-invalid-border: 1px solid #{$sky-highlight-color-danger};
+  --sky-override-file-attachment-invalid-box-shadow: 0 0 8px
+    rgba(239, 64, 68, 0.6);
+  --sky-override-file-attachment-invalid-label-color: #{$sky-theme-modern-font-deemphasized-color};
+  --sky-override-file-attachment-preview-img-border: var(
+      --sky-border-width-separator-dark
+    )
+    solid var(--sky-color-border-separator-dark);
+  --sky-override-file-attachment-preview-img-box-shadow: none;
+  --sky-override-file-attachment-preview-img-margin: 0 #{$sky-theme-modern-space-sm}
+    0 0;
+  --sky-override-file-attachment-reject-border-radius: 5px;
+  --sky-override-file-attachment-reject-border-width: 1px;
+
+  .sky-file-attachment-delete {
+    padding: 0;
+  }
+}
 
-:host.sky-margin-stacked-lg {
+@include compatMixins.sky-default-host-overrides() {
+  --sky-override-file-attachment-hint-text-margin-top: var(
+    --sky-margin-stacked-xs
+  );
+}
+
+:host.sky-form-field-stacked {
   display: block;
 }
 
+:host:not(.sky-form-field-stacked) {
+  .sky-file-attachment-wrapper {
+    .sky-file-attachment-btn {
+      margin: var(--sky-override-file-attachment-btn-non-stacked-margin, 0);
+    }
+  }
+}
+
 .sky-file-attachment-wrapper {
-  display: inline-block;
+  display: var(--sky-override-file-attachment-wrapper-display, flex);
+  flex-wrap: wrap;
+}
+
+.sky-file-attachment-label-wrapper {
+  width: 100%;
 }
 
 .sky-file-attachment-upload {
-  padding-right: 8px;
+  flex: 1;
+  order: 2;
+  display: flex;
+  align-items: flex-start;
+  padding-right: var(--sky-override-file-attachment-upload-padding-right, 0);
   display: block;
   border: 1px solid transparent;
 }
 
 .sky-file-attachment-accept {
-  border-color: $sky-highlight-color-success;
+  border-color: var(
+    --sky-override-file-attachment-accept-border-color,
+    var(--sky-color-border-success)
+  );
   border-style: dashed;
-  border-width: 1px;
-  border-radius: 5px;
-  color: $sky-text-color-default;
+  border-width: var(
+    --sky-override-file-attachment-accept-border-width,
+    var(--sky-border-width-input-hover)
+  );
+  border-radius: var(
+    --sky-override-file-attachment-accept-border-radius,
+    var(--sky-border-radius-s)
+  );
 }
 
 .sky-file-attachment-reject {
-  border-color: $sky-highlight-color-danger;
+  border-color: var(
+    --sky-override-file-attachment-reject-border-color,
+    var(--sky-color-border-danger)
+  );
   border-style: dashed;
-  border-width: 1px;
-  border-radius: 5px;
-  color: $sky-text-color-default;
+  border-width: var(
+    --sky-override-file-attachment-reject-border-width,
+    var(--sky-border-width-input-hover)
+  );
+  border-radius: var(
+    --sky-override-file-attachment-reject-border-radius,
+    var(--sky-border-radius-s)
+  );
+}
+
+.sky-file-attachment-btn,
+.sky-control-label {
+  transition: $sky-form-border-and-color-transitions;
 }
 
-:host(.ng-invalid.ng-dirty) {
+:host(.ng-invalid.ng-touched) {
```

Comment:
This is a bug fix that I slipped in. Previously, the error message for errors was showing up on touched but the error border on the button was not.

Reply:
Actually - I reverted this. It was causing some visual failures and I'd rather do it in a separate PR.